src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -292,19 +291,17 @@ public Object run() {
             }
         });
 
-        advancedModulesEnabled = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_ADVANCED_MODULES_ENABLED, true);
-        ReflectionHelper.init(advancedModulesEnabled);
-        
-        ReflectionHelper.registerMngtRestApiHandler(settings);
+        final String advancedModulesEnabledKey = ConfigConstants.OPENDISTRO_SECURITY_ADVANCED_MODULES_ENABLED;
+        if (settings.hasValue(advancedModulesEnabledKey)) {
+            log.warn("Setting {} is ignored and is enabled by default if plugin is not running in SSL only or disabled mode.", advancedModulesEnabledKey);
```

Comment:
Consider using `DeprecationLogger`. IMO, it will be just sufficient to notify that the setting is ignored.